docs/blueprints/forgotten-deletes.md

```diff
@@ -0,0 +1,124 @@
+    Simply leaving the value blank doesn't really solve the problem either, as 0 is a specific version already. So we would need to pass along another value in the code, perhaps a boolean indicating that this is deleted piece data, in addition to the format version in order to process deleted piece data correctly, but I have a feeling this approach will result in clutter. Here's a hacky idea: what if we did define a new format version for deleted piece data, but set it to -1? This would circumvent some of the problems with format versions. Though, I'm not sure if this approach holds any substantial benefits over using a separate directory.
+
+## Implementation
+
+1. Add new field, satellite_signature, to DeletePiecesRequest protobuf
+
+2. Add new field, delete_pieces_request, to PieceDownloadResponse protobuf
+
+3. Add code for satellite to sign DeletePiecesRequests
+
+4. Implement file store for piece IDs and signed delete requests
+
+5. Add code to check for and return PieceDeleteRequest if piece cannot be found during download
+
+6. Add code to satellite to delete object if PieceDownloadResponse contains a valid DeletePiecesRequest
+
+7. Determine the furtherst point in the past we are allowed to revert to. If we are not already doing so, we need to automate taking snapshots of the pointer DB and make sure we run garbage collection on the appropriate one.
+
+8. Garbage collection retain messages currently set creationDate to the current time. This value is used to determine which pieces on the storage node should not be deleted because they are new and were not seen by the piece tracker. This value will need to be set to the time of the DB snapshot on which we are running garbage collection. Otherwise, the storage nodes will delete all pieces created since the DB snapshot.
+
+## Wrapup
+
+- We may need to edit documents regarding piece deletion/garbage collection
+
+## Open issues
+
+- The design laid out in this document addresses the issue of falsely punishing nodes due to a DB restoration. It does not however, solve all data loss problems that could arise from a DB restoration. For example, how will we handle forgotten uploads?
+
+- The solution to forgotten garbage collection deletes is going to increase the TTL of trash that nodes hold. They will probably not like that. The extent to which the TTL of garbage is increased depends on how far back in time we may want to restore.
```

Comment:
With WAL files, I would hope that any restoration would have minimal effect. However, if we are using the most recent base-backup DB for garbage collection, it would not include the WAL files. In that case, it would probably delay GC by 1 week which definitely isn't great :/
I wonder if there is a way to include the WAL files before we iterate.